src/auth/README.md

```diff
@@ -72,69 +83,196 @@ if (blockstack.isUserSignedIn()) {
 
-```es6
-import { makeAuthResponse, makeECPrivateKey } from 'blockstack'
+### Blockstack ID Identity address private key
+The identity address private key is derived from the user's keychain phrase and is the private key of the Blockstack ID that the user chooses to use to sign in to the app. It is a secret owned by the user and never leaves the user's instance of the Blockstack browser. This private key signs the authentication response token for an app to indicate that the user approves sign in to that app.
 
-const privateKey = makeECPrivateKey()
-const profile = { "name": "Naval Ravikant" }
-const username = "naval.id"
-const authResponse = makeAuthResponse(privateKey, profile, username)
+### App private key
+
+The app private key is an app-specific private key that is generated from the user's identity address private key using the `domain_name` as input. It is deterministic in that for a given Blockstack ID and `domain_name`, the same private key will be generated each time. The app private key is securely shared with the app on each authentication, encrypted by the Blockstack browser with the transit public key.  
+
+The app private key serves three functions. 
+* It is used to create the credentials that give an app access to the gaia hub storage bucket for that specific app. 
+* It is used in the end-to-end encryption of files stored for the app on the user's gaia hub. 
+* It serves as a cryptographic secret that apps can use to perform other cryptographic functions.
+
+## Scopes
+Scopes define the information and permissions an app requests from the user during authentication. If no `scopes` array is provided to the `redirectToSignIn` or `makeAuthRequest` functions.
+- `store_write` - read and write data to the user's Gaia hub in an app-specific storage bucket
+- `publish_data` - publish data so that other users of the app can discover and interact with the user
+- `email` - requests the user's email if available
+
+## Authentication tokens
+
+Communication between the app requesting authentication and the Blockstack Browser, the component that authenticates the user, happens via a protocol based on passing JSON web tokens via the URL query string. There are two types of authentication tokens, the authentication request token which is generated by an app requesting authentication of a user and sent to the user's Blockstack Browser and the authentication response token which is sent from the Blockstack Browser to the app when the user approved sign in to the app.
```

Comment:
The app and the Blockstack Browser communicate during the authentication flow by passing back and forth two tokens:

1. The requesting application sends the Blockstack Browser an `authRequest` token.
2. Once a user approves a sign in, the Blockstack Browser responds to the application with an `authResponse` token.

These tokens are [JSON Web Tokens](https://jwt.io/), and they are passed via URL query strings.

Reply:
Much better wording. Thanks 🥇 !